### ISSUE TITLE
[BUG] Impossible to actually import codes

### ISSUE THREAD

**Issue by RyanHakurei** [0x4f53/Wristkey#16]
**Describe the bug**
It's less a bug and more of a design flaw. The directory the app looks for backups in is not actually accessible to the user.

**To Reproduce**
1. Attempt to push the vault export via adb
2. Permission denied

**Expected behavior**
I expect the app to use a location that the user actually has permissions to, such as a folder under `/sdcard`

**Screenshots**
![Screenshot_20230507_103926_wristkey](https://user-images.githubusercontent.com/5086636/236684647-2a7191f1-2682-4414-95c4-32c0f0c3d1c3.png)
![Screenshot - 2023-05-07 10-10](https://user-images.githubusercontent.com/5086636/236684112-4c546cf8-82fd-4a41-8454-b126d00882a9.png)

**Watch:**
 - Watch Name and model number: Galaxy Watch 4
 - OS: WearOS 3.5
 - Wristkey version 2.2

**Smartphone (if applicable):**

**Additional context**
There is currently no workaround it seems so this app is unusable on WearOS as it stands as WearOS lacks the Storage Access Framework, so even sideloading an app such as Solid Explorer will not allow you to pick a file.

**Comment by 4f77616973** [comment]
This is must be a new thing because I never got this issue in my testing. I haven't worked on this project for a while, so I will have to take some time in the near future to write a proper phone client to enable imports...

In the meantime, check out #10 . Another user had issues importing data, and turns out, they needed to use `adb shell` to make the `files/` directory first, then push with a slash appended to the end of the path.

**Comment by Marckus** [comment]
> **Describe the bug** It's less a bug and more of a design flaw. The directory the app looks for backups in is not actually accessible to the user.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
>     1. Attempt to push the vault export via adb
> 
>     2. Permission denied
> 
> 
> **Expected behavior** I expect the app to use a location that the user actually has permissions to, such as a folder under `/sdcard`
> 
> **Screenshots** ![Screenshot_20230507_103926_wristkey](https://user-images.githubusercontent.com/5086636/236684647-2a7191f1-2682-4414-95c4-32c0f0c3d1c3.png) ![Screenshot - 2023-05-07 10-10](https://user-images.githubusercontent.com/5086636/236684112-4c546cf8-82fd-4a41-8454-b126d00882a9.png)
> 
> **Watch:**
> 
>     * Watch Name and model number: Galaxy Watch 4
> 
>     * OS: WearOS 3.5
> 
>     * Wristkey version 2.2
> 
> 
> **Smartphone (if applicable):**
> 
>     * Device: [e.g. iPhone6]
> 
>     * OS: [e.g. iOS8.1]
> 
> 
> **Additional context** There is currently no workaround it seems so this app is unusable on WearOS as it stands as WearOS lacks the Storage Access Framework, so even sideloading an app such as Solid Explorer will not allow you to pick a file.

I have this same issue with a GW4 as well.  There is no way to access the directory you reference.  Also, in following all of your instructions I was able to verify that a screenshot of my Google Auth QR Codes was pushed to the appropriate directory on the watch.  However, when I got to import the QR Codes, I get an error message on the watch that "This appears to be a Google Authenticator export.  Pleas..." and I cannot see the rest of the message.  I'm assuming it says something along the lines of please be sure the image is in focus, etc., the issue is, when I run your app on my phone scanning in that same image file with my phone's camera, it works fine.  I can't copy the image file to the location on my phone to test it that way, due to this same design flaw unfortunately.  I do appreciate all the work you've done to make this though!

**Comment by 4f77616973** [comment]
@Marckus Try #10 as well and tell me how it goes. Use `adb shell` to make a `files/` directory and push it like that.

I have an LG G Watch (dory) and the latest Wear OS AVD on my computer and don't face these issues. If this is some new thing that Samsung/Google added, then I'm sorry, but I can't do anything to fix it.

**Comment by 4f77616973** [comment]
Android 11 (which is what the latest Wear OS is based on) uses [scoped storage](https://source.android.com/docs/core/storage/scoped). I updated Wristkey to work on any device, including phones and non-Wear OS AOSP-based watches. It is possible that Google updated Wear OS to ignore this storage thing and completely eliminated user disk-access via the storage framework.

This might explains why it works perfectly on my original G Watch (running the lenok2dory Android 7.1 custom ROM) and a Wear OS AVD but not an actual watch. 

No worries though. Now that the app works on both the phone and the watch, I just realized it's even easier to write a little component that simply detects if the device is Wear OS and allows it to receive data from a phone. Got a lot going on at the moment. When I get some free time, I'll try making a phone companion.
